### PR TITLE
A bunch of enhancements and fixes regarding the Block Pocket

### PR DESCRIPTION
**Pull request by RedstoneDubstep**
In the last few days I thought about the Block Pocket and how to enhance this already great feature. No, you still cannot disguise the BP with every block you want (I'll leave the design decision of this enhancement to the devs), but now you're just one click (and a massive, massive grind -to gather 7682 prismarine shards is quite a ton of farming) away from building a 25x25x25 Block Pocket without placing more than one block for yourself!

### CODE REVIEW

File: src/main/java/net/geforcemods/securitycraft/screen/BlockPocketManagerScreen.java [hunk 1]
```diff
@@ -53,6 +58,29 @@ protected void drawGuiContainerForegroundLayer(int mouseX, int mouseY)
 		String translation = ClientUtils.localize(SCContent.BLOCK_POCKET_MANAGER.get().getTranslationKey());
 
 		font.drawString(translation, xSize / 2 - font.getStringWidth(translation) / 2, 6, 4210752);
+
+		if (!te.enabled)
+		{
+			font.drawString(ClientUtils.localize("gui.securitycraft:blockPocketManager.youNeed"), 7, 83, 4210752);
+
+			font.drawString((size - 2) * (size - 2) * 6 + "", 42, 100, 4210752);
+			minecraft.getItemRenderer().renderItemAndEffectIntoGUI(new ItemStack(SCContent.BLOCK_POCKET_WALL.get()), 25, 96);
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/screen/BlockPocketManagerScreen.java:67]: Please don't create a new ItemStack every tick. Instead, create private static final fields at the top of this class that contain all the ItemStacks you use in this method, and use those fields instead.

File: src/main/java/net/geforcemods/securitycraft/screen/BlockPocketManagerScreen.java [hunk 2]
```diff
@@ -91,6 +119,27 @@ public void sizeButtonClicked(ClickButton button)
...
+		if(feedback != null)
+			PlayerUtils.sendMessageToPlayer(Minecraft.getInstance().player, ClientUtils.localize(SCContent.BLOCK_POCKET_MANAGER.get().getTranslationKey()), ClientUtils.localize(feedback.getKey(), feedback.getFormatArgs()), TextFormatting.DARK_AQUA);
+
+		Minecraft.getInstance().player.closeScreen();
+	}
+
+	public void outlineButtonClicked(ClickButton button)
+	{
+		te.toggleOutline();
+
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/screen/BlockPocketManagerScreen.java:142]: Unnecessary new line.

File: src/main/java/net/geforcemods/securitycraft/tileentity/BlockPocketManagerTileEntity.java [hunk 3]
```diff
@@ -204,6 +225,430 @@ else if((xi == lowest || xi == highest) && zi > lowest && zi < highest)
...
 
+	/**
+	 * Auto-assembles the Block Pocket for a player.
+	 * First it makes sure that the space isn't occupied, then
+	 * it places the blocks.
+	 * @param player The player that opened the screen, used to
+	 *               assign the owner to the placed blocks
+	 * @return The feedback message. null if none should be sent.
+	 */
+
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/tileentity/BlockPocketManagerTileEntity.java:236]: Unnecessary empty line

File: src/main/java/net/geforcemods/securitycraft/tileentity/BlockPocketManagerTileEntity.java [hunk 4]
```diff
@@ -204,6 +225,430 @@ else if((xi == lowest || xi == highest) && zi > lowest && zi < highest)
 		return null;
 	}
 
+	/**
+	 * Auto-assembles the Block Pocket for a player.
+	 * First it makes sure that the space isn't occupied, then
+	 * it places the blocks.
+	 * @param player The player that opened the screen, used to
+	 *               assign the owner to the placed blocks
+	 * @return The feedback message. null if none should be sent.
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/tileentity/BlockPocketManagerTileEntity.java:234]: Unnecessary line breaks.

File: src/main/java/net/geforcemods/securitycraft/tileentity/BlockPocketManagerTileEntity.java [hunk 5]
```diff
@@ -204,6 +225,430 @@ else if((xi == lowest || xi == highest) && zi > lowest && zi < highest)
...
+	 * First it makes sure that the space isn't occupied, then
+	 * it places the blocks.
+	 * @param player The player that opened the screen, used to
+	 *               assign the owner to the placed blocks
+	 * @return The feedback message. null if none should be sent.
+	 */
+
+    public TranslationTextComponent autoAssembleMultiblock(PlayerEntity player)
+    {
+        if(!enabled) //multiblock assembling in three steps
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/tileentity/BlockPocketManagerTileEntity.java:239]: 4 spaces -> 1 tab

File: src/main/java/net/geforcemods/securitycraft/tileentity/BlockPocketManagerTileEntity.java [hunk 6]
```diff
@@ -204,6 +225,430 @@ else if((xi == lowest || xi == highest) && zi > lowest && zi < highest)
...
+					pos = startingPos.up(yi).offset(back, zi);
+				}
+				zi = 0;
+				yi++;
+				pos = startingPos.up(yi);
+			}
+			return new TranslationTextComponent("messages.securitycraft:blockpocket.assembled");
+		}
+        return null;
+    }
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/tileentity/BlockPocketManagerTileEntity.java:650]: 4 spaces -> 1 tab

File: src/main/java/net/geforcemods/securitycraft/screen/BlockPocketManagerScreen.java (hunk 1 above)
Comment (bl4ckscor3) [reply src/main/java/net/geforcemods/securitycraft/screen/BlockPocketManagerScreen.java]: There are still some ItemStacks further down.

File: src/main/resources/assets/securitycraft/lang/de_de.json [hunk 7]
```diff
@@ -592,6 +592,10 @@
 	"gui.securitycraft:blockPocketManager.activate": "Aktivieren",
 	"gui.securitycraft:blockPocketManager.deactivate": "Deaktivieren",
 	"gui.securitycraft:blockPocketManager.size": "Größe: %sx%sx%s",
+	"gui.securitycraft:blockPocketManager.assemble": "Auto-Aufbauen",
+	"gui.securitycraft:blockPocketManager.outline.show": "Umrandung zeigen",
+	"gui.securitycraft:blockPocketManager.outline.hide": "Umrandung verbergen",
+	"gui.securitycraft:blockPocketManager.youNeed": "Du brauchst:",
```
Comment (bl4ckscor3) [inline src/main/resources/assets/securitycraft/lang/de_de.json:598]: I think `Du benötigst:` sounds better.

File: src/main/java/net/geforcemods/securitycraft/screen/BlockPocketManagerScreen.java [hunk 8]
```diff
@@ -24,6 +25,11 @@
 	private int size = 5;
 	private Button toggleButton;
 	private Button sizeButton;
+	private Button assembleButton;
+	private Button outlineButton;
+	private static final ItemStack blockPocketWall = new ItemStack(SCContent.BLOCK_POCKET_WALL.get());
+	private static final ItemStack chiseledCrystalQuartz = new ItemStack(SCContent.REINFORCED_CHISELED_CRYSTAL_QUARTZ.get());
+	private static final ItemStack crystalQuartzPillar = new ItemStack(SCContent.REINFORCED_CRYSTAL_QUARTZ_PILLAR.get());
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/screen/BlockPocketManagerScreen.java:32]: Names of static final fields (=constants) are written in upper snake case by convention. Example: `BLOCK_POCKET_WALL`. The names of the latter two variables also don't reflect the block that the ItemStack consists of. Please change.

File: src/main/resources/assets/securitycraft/lang/de_de.json (hunk 7 above)
Comment (RedstoneDubstep) [reply src/main/resources/assets/securitycraft/lang/de_de.json:598]: Right.

File: src/main/java/net/geforcemods/securitycraft/screen/BlockPocketManagerScreen.java [hunk 9]
```diff
@@ -53,6 +61,29 @@ protected void drawGuiContainerForegroundLayer(int mouseX, int mouseY)
...
+
+		if (!te.enabled)
+		{
+			font.drawString(ClientUtils.localize("gui.securitycraft:blockPocketManager.youNeed"), 7, 83, 4210752);
+
+			font.drawString((size - 2) * (size - 2) * 6 + "", 42, 100, 4210752);
+			minecraft.getItemRenderer().renderItemAndEffectIntoGUI(BLOCK_POCKET_WALL, 25, 96);
+
+			font.drawString((size - 2) * 12 - 1 + "", 94, 100, 4210752);
+			minecraft.getItemRenderer().renderItemAndEffectIntoGUI(REINFORCED_CHISELED_CRYSTAL_QUARTZ, 77, 96);
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/screen/BlockPocketManagerScreen.java:73]: You mixed up the ItemStack here with the one on line 76.

File: src/main/java/net/geforcemods/securitycraft/screen/BlockPocketManagerScreen.java [hunk 10]
```diff
@@ -53,6 +61,29 @@ protected void drawGuiContainerForegroundLayer(int mouseX, int mouseY)
...
+			minecraft.getItemRenderer().renderItemAndEffectIntoGUI(REINFORCED_CRYSTAL_QUARTZ_PILLAR, 130, 96);
+
+			if(mouseX >= guiLeft + 23 && mouseX < guiLeft + 48 && mouseY >= guiTop + 93 && mouseY < guiTop + 115)
+				renderTooltip(new ItemStack(SCContent.BLOCK_POCKET_WALL.get()), mouseX - guiLeft, mouseY - guiTop);
+
+			if(mouseX >= guiLeft + 75 && mouseX < guiLeft + 100 && mouseY >= guiTop + 93 && mouseY < guiTop + 115)
+				renderTooltip(new ItemStack(SCContent.REINFORCED_CRYSTAL_QUARTZ_PILLAR.get()), mouseX - guiLeft, mouseY - guiTop);
+
+			if(mouseX >= guiLeft + 128 && mouseX < guiLeft + 153 && mouseY >= guiTop + 93 && mouseY < guiTop + 115)
+				renderTooltip(new ItemStack(SCContent.REINFORCED_CHISELED_CRYSTAL_QUARTZ.get()), mouseX - guiLeft, mouseY - guiTop);
```
Comment (bl4ckscor3) [inline src/main/java/net/geforcemods/securitycraft/screen/BlockPocketManagerScreen.java:85]: There are _still_ 3 instances here where you create a new ItemStack each tick (on hover, but still).

Comment (RedstoneDubstep) [reply src/main/java/net/geforcemods/securitycraft/screen/BlockPocketManagerScreen.java]: Sorry, I didn't see these instances and I also didn't look at the updated review above, so it took me some time.

File: src/main/resources/assets/securitycraft/lang/en_us.json [hunk 11]
```diff
@@ -664,10 +668,15 @@
 	"messages.securitycraft:sentry.descriptionMode3": "(Never active, doesn't attack anything)",
 	"messages.securitycraft:blockpocket.multipleManagers": "There is more than one Block Pocket Manager in the structure.",
 	"messages.securitycraft:blockpocket.invalidBlock": "Invalid block \"%2$s\" at position %1$s!",
-	"messages.securitycraft:blockpocket.invalidBlock.pillar": "Invalid block \"%2$s\" at position %1$s! (Make sure the pillar is rotated correctly!)",
+	"messages.securitycraft:blockpocket.invalidBlock.pillar": "Invalid block \"%2$s\" at position %1$s! (Expected \"Reinforced Crystal Quartz Pillar\")",
+	"messages.securitycraft:blockpocket.invalidBlock.pillar.rotation": "Invalid block \"%2$s\" at position %1$s! (Make sure the pillar is rotated correctly!)",
+	"messages.securitycraft:blockpocket.invalidBlock.corner": "Invalid block \"%2$s\" at position %1$s! (Expected \"Reinforced Chiseled Crystal Quartz Block\")",
+	"messages.securitycraft:blockpocket.invalidBlock.wall": "Invalid block \"%2$s\" at position %1$s! (Expected \"Block Pocket Wall\")",
```
Comment (bl4ckscor3) [inline src/main/resources/assets/securitycraft/lang/en_us.json:674]: Needs a colon after `Expected`

**Pull request merged**